components/aws/sagemaker/common/boto3_manager.py

```diff
@@ -112,6 +113,73 @@ def get_sagemaker_client(
         )
         return client
 
+    @staticmethod
+    def get_sagemaker_session(
+        component_version: str,
+        region: str,
+        endpoint_url: str = None,
+        assume_role_arn: str = None,
+    ):
```

Comment:
I guess it would cease to be a staticmethod at that point. We could, but that might mean changing the get_sagemaker_client method to return both the sagemaker client it instantiates and also the boto3 session that it creates. We would then need to update everywhere get_sagemaker_client method is called to ignore the returned session that is not being used elsewhere. Open to suggestions, its not very DRY but seems like a simple solution as SageMakerSession is not used elsewhere

Reply:
Updated as per recommendation